src/array.rs

```diff
@@ -110,7 +111,7 @@ impl<'a, const P: usize, const W: usize> RepresentationTrait for Array<'a, P, W>
     /// Return memory size of `Array` representation
     #[inline]
     fn size_of(&self) -> usize {
-        size_of::<usize>() + size_of_val(self.arr)
+       size_of::<usize>() + size_of::<usize>() + size_of::<&[u32]>() + size_of_val(self.arr)
```

Comment:
I'm not sure about this `size_of` calculation logic. Originally, I was counting it as sum of:
* data: usize - which stores either data or pointer to the data
* optional u32 slice

Reply:
Let me comment it better

Reply:
Done that in commit [20f3d0d](https://github.com/cloudflare/cardinality-estimator/pull/5/commits/20f3d0dae360e4c73f7f3d714d5ff63fcbea2b6b), let me know if all is clear.